WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -69,67 +88,142 @@ public Wallet GetFirstOrDefaultWallet()
 		{
 			lock (Lock)
 			{
-				return Wallets.Keys.FirstOrDefault();
+				return Wallets.Keys.FirstOrDefault(x => x.State == WalletState.Started);
 			}
 		}
 
 		public bool AnyWallet()
 		{
 			lock (Lock)
 			{
-				return Wallets.Keys.Any();
+				return Wallets.Keys.Any(x => x.State >= WalletState.Starting);
 			}
 		}
 
-		public async Task<Wallet> CreateAndStartWalletAsync(KeyManager keyManager)
+		public async Task<Wallet> CreateAndStartWalletAsync(string walletName)
+		{
+			KeyManager keyManager;
+			lock (Lock)
+			{
+				keyManager = Wallets.Single(x => x.Key.KeyManager.GetName() == walletName).Key.KeyManager;
+			}
+
+			return await CreateAndStartWalletAsync(keyManager).ConfigureAwait(false);
```

Comment:
This method don't need to be marked as `async` and could just return the tasks returned by `CreateAndStartWalletAsync(keyManager)`
```suggestion
			return CreateAndStartWalletAsync(keyManager);
```